common/src/main/java/com/wynntils/functions/HorseFunctions.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.functions;
+
+import com.wynntils.core.functions.ActiveFunction;
+import com.wynntils.mc.utils.InventoryUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wc.custom.item.WynnItemStack;
+import com.wynntils.wc.custom.item.properties.ItemProperty;
+import net.minecraft.world.item.ItemStack;
+
+public class HorseFunctions {
+
+    private static WynnItemStack getHorse() {
+        if (InventoryUtils.findHorseSlotNum() == -1) return null;
+        ItemStack is = McUtils.player().getInventory().getItem(InventoryUtils.findHorseSlotNum());
+        if (!(is instanceof WynnItemStack horse)) return null;
+        return horse;
+    }
+
+    public static class HorseLevelFunction extends ActiveFunction<String> {
+        @Override
+        public String getValue(String argument) {
+            if (getHorse() == null) return "??";
```

Comment:
Please store the result of getHorse() in a local variable, and use that instead. There is an optimization win (we don't have to go through all the inventory slots twice to locate the horse), but the big win is in stability. If something where to change in the background, we could fail even if you did check previously.

In general, this is something you always should do in programming. It's called [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) -- "time of check to time of use" and is a classical source of race condition.